File: src/main/java/com/github/alexeysol/geekregimeapiposts/ApiUsersSourceResolver.java
package com.github.alexeysol.geekregimeapiposts;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class ApiUsersSourceResolver {
    @Value("${api-users.base-url}")
    private String baseUrl;

    @Value("${api-users.prefix}")
    private String apiPrefix;

    @Value("${api-users.resource}")
    private String resource;

    public String getBaseUrl() {
        return baseUrl;
    }

    public String getApiPath(int version) {
        return String.format("%s/v%d/%s", apiPrefix, version, resource);
    }

    public String getApiPath() {
        return getApiPath(1);
    }
}

File: src/test/kotlin/com/github/alexeysol/geekregimeapiusers/ApiUsersSourceResolver.kt
package com.github.alexeysol.geekregimeapiusers

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
data class ApiUsersSourceResolver (
    @Value("\${api-users.prefix}")
    val apiPrefix: String? = null,

    @Value("\${api-users.resource}")
    val resource: String? = null,
) {
    fun getApiPath(version: Int = 1) = "/${apiPrefix}/v${version}/${resource}"
}
